akka-projection-grpc/src/main/scala/akka/projection/grpc/consumer/scaladsl/EventProducerPushDestination.scala

```diff
@@ -75,24 +81,30 @@ object EventProducerPushDestination {
       val mapTags = { (eventEnvelope: EventEnvelope[Any]) =>
         val newTags = f(eventEnvelope.asInstanceOf[EventEnvelope[A]])
         if (newTags eq eventEnvelope.tags) eventEnvelope
-        else copyEnvelope(eventEnvelope)(tags = newTags)
+        else eventEnvelope.withTags(newTags)
       }
-      appendMapper(clazz, mapTags)
+      appendMapper(clazz, mapTags, payloadMapper = false)
     }
 
     /**
      * Transform incoming persistence ids, care must be taken to produce a valid persistence id and to always map the
      * same incoming persistence id to the same stored persistence id to not introduce gaps in the sequence numbers
      * and break consuming projections.
      */
-    def registerPersistenceIdMapper(f: EventEnvelope[Any] => String): Transformation = {
+    def registerPersistenceIdMapper(f: EventEnvelope[Any] => String)(
+        implicit system: ActorSystem[_]): Transformation = {
```

Comment:
had to add the ActorSystem here because need the sliceForPersistenceId from Persistence extension